packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx

```diff
@@ -80,13 +88,33 @@ export const ActionBarItem = ({ item }: ActionBarItemProps) => {
           }
         />
       ) : (
-        <StyledButton
-          accent={item.accent ?? 'default'}
-          onClick={() => item.onClick?.()}
-        >
-          {item.Icon && <item.Icon size={theme.icon.size.md} />}
-          <StyledButtonLabel>{item.label}</StyledButtonLabel>
-        </StyledButton>
+        <>
+          <StyledButton
+            accent={item.accent ?? 'default'}
+            onClick={() =>
+              item.label === 'Delete'
+                ? setIsDeleteRecordsModalOpen(true)
+                : item.onClick?.()
+            }
+          >
+            {item.Icon && <item.Icon size={theme.icon.size.md} />}
+            <StyledButtonLabel>{item.label}</StyledButtonLabel>
+          </StyledButton>
+          {item.label === 'Delete' && (
+            <ConfirmationModal
+              isOpen={isDeleteRecordsModalOpen}
+              setIsOpen={setIsDeleteRecordsModalOpen}
+              title={`Delete ${selectedRecords} ${
+                selectedRecords === 1 ? `record` : 'records'
+              }`}
+              subtitle={`This action cannot be undone. This will permanently delete ${
+                selectedRecords === 1 ? 'this record' : 'these records'
+              }`}
+              onConfirmClick={() => item.onClick?.()}
+              deleteButtonText="Delete Records"
+            />
+          )}
+        </>
```

Comment:
This doesn't seem to be a good approach as the base action for Delete is already implemented and conditional approach isn't good. We can keep the ActionBarItem code as it is. Instead please make changes to `useRecordActionBar`

Reply:
tbh, i had doubts regarding my implementation. So, thanks for the review. I have made the necessary changes. Is this what you meant?

Reply:
Yes basically I was suggesting reusability and not hard coded.